scipy/optimize/_linprog_rs.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Aug 18 22:07:15 2018
+
+@author: matthaberland
+"""
+
+import numpy as np
+from scipy.linalg import solve
+from .optimize import _check_unknown_options
+from ._bglu_dense import LU
+from ._bglu_dense import BGLU as BGLU
+from scipy.linalg import LinAlgError
+from numpy.linalg.linalg import LinAlgError as LinAlgError2
+
+# TODO: fix test_bug_8662. Bug is that auxiliary problem terminates with
+# artificial variable in basis; that is discarded and a linearly depended
+# column is chosen to replace it.
+
+# TODO: fix test_bug_6690. Bug is that auxiliary problem terminates with
+# artificial variable in basis; that is discarded and a linearly depended
+# column is chosen to replace it. Regular simplex is successful.
+
+# TODO: fix test_bug_7044. Bug is that auxiliary problem terminates with
+# a singular basis. Doesn't happen with small number of updates, but does with
+# exact solve.
+
+# TODO: fix test_bug_5400. Bug is that in iteration 8 pivots to a singular
+# basis. Current default behavior is status 4; better than simplex which is
+# status 2. When perfect solve is used, no error status.
+
+# TODO: fix test_large_problem. Bug is that in iteration 199 pivots to a
+# singular basis. Doesn't happen with small number of updates.
+
+# TODO: add display?
+
+
+def _phase_one(A, b, maxiter, tol, maxupdate, mast, pivot):
```

Comment:
This function, and some others below, repeats behaviour already used in ``simplex``,  but also checks for redundancies. Would it be worth keeping all the simplex methods in a single module?

Reply:
This has been discussed in the comments